yb-voyager/cmd/importDataToSource.go

```diff
@@ -112,7 +114,7 @@ func packAndSendImportDataToSourcePayload(status string, errorMsg string) {
 
 	importDataPayload.Phase = importPhase
 
-	if importPhase != dbzm.MODE_SNAPSHOT {
+	if importPhase != dbzm.MODE_SNAPSHOT && statsReporter != nil {
```

Comment:
what is the case when statsReporter will be null?

Reply:
It can be nil until it is initialized. If there is any failure or exit before that, this function will try to send the payload, and we can get a nil pointer exception in such a case.

Reply:
https://www.honeybadger.io/blog/go-exception-handling/

Reply:
@sanyamsinghal if I understood you correctly, you're suggesting that we have some sort of a try-catch for the exit handlers as a safety net so that they don't panic, is that correct? 

If so, atexit already sort of does that. Check this out:
```
func runHandler(handler func()) {
	defer func() {
		if err := recover(); err != nil {
			fmt.Fprintln(os.Stderr, "error: atexit handler error:", err)
		}
	}()

	handler()
}
```

Reply:
Yeah right, in this case I am fixing in this PR, it wasn't panicking with stack trace, it was just erroring with below msg
```
error: atexit handler error: runtime error: invalid memory address or nil pointer dereference
```

Reply:
@makalaaneesh @priyanshi-yb  what i was recommending is that we don't even error out if callhome payload sending failing.
Maybe just log that error, or max to max display on console.

Any failure in sending payload to callhome should not result in an error is what I think

Reply:
> Maybe just log that error, or max to max display on console.

exit handler also displays the msg on the console for any failure, and the issue we hit is for the exit case only.

Reply:
maybe i am missing something, lets discuss this overall tomorrow.